packages/fetchai/skills/oracle_aggregation/handlers.py

```diff
@@ -0,0 +1,283 @@
+
+
+class GenericConsensusHandler(Handler):
+    """This class implements a Consensus handler."""
+
+    SUPPORTED_PROTOCOL = ConsensusMessage.protocol_id  # type: Optional[PublicId]
+
+    def setup(self) -> None:
+        """Implement the setup for the handler."""
+        pass
+
+    def handle(self, message: Message) -> None:
+        """
+        Implement the reaction to a message.
+
+        :param message: the message
+        :return: None
+        """
+        consensus_msg = cast(ConsensusMessage, message)
+
+        # recover dialogue
+        consensus_dialogues = cast(ConsensusDialogues, self.context.consensus_dialogues)
+        consensus_dialogue = cast(
+            ConsensusDialogue, consensus_dialogues.update(consensus_msg)
+        )
+        if consensus_dialogue is None:
+            self._handle_unidentified_dialogue(consensus_msg)
+            return
+
+        # handle message
+        if consensus_msg.performative == ConsensusMessage.Performative.OBSERVATION:
+            self._handle_observation(consensus_msg)
+        elif consensus_msg.performative == ConsensusMessage.Performative.AGGREGATION:
+            self._handle_aggregation(consensus_msg)
+        else:
+            self._handle_invalid(consensus_msg, consensus_dialogue)
+
+    def teardown(self) -> None:
+        """
+        Implement the handler teardown.
+
+        :return: None
+        """
+        pass
+
+    def _handle_unidentified_dialogue(self, consensus_msg: ConsensusMessage) -> None:
+        """
+        Handle an unidentified dialogue.
+
+        :param consensus_msg: the message
+        """
+        self.context.logger.info(
+            "received invalid consensus message={}, unidentified dialogue.".format(
+                consensus_msg
+            )
+        )
+        default_dialogues = cast(DefaultDialogues, self.context.default_dialogues)
+        default_msg, _ = default_dialogues.create(
+            counterparty=consensus_msg.sender,
+            performative=DefaultMessage.Performative.ERROR,
+            error_code=DefaultMessage.ErrorCode.INVALID_DIALOGUE,
+            error_msg="Invalid dialogue.",
+            error_data={"consensus_message": consensus_msg.encode()},
+        )
+        self.context.outbox.put_message(message=default_msg)
+
+    def _handle_observation(self, obs_msg: ConsensusMessage) -> None:
+        """
+        Handle the observation.
+
+        :param obs_msg: the message
+        :param consensus_dialogue: the dialogue object
+        :return: None
+        """
+
+        self.context.logger.info(
+            "received observation from sender={}".format(obs_msg.sender[-5:])
+        )
+
+        strategy = cast(GenericStrategy, self.context.strategy)
+        obs = {
+            "value": obs_msg.value,
+            "time": obs_msg.time,
+            "source": obs_msg.source,
+            "signature": obs_msg.signature,
+        }
+
+        strategy.add_observation(obs_msg.sender, obs)
```

Comment:
```suggestion
        obs = {
            "value": obs_msg.value,
            "time": obs_msg.time,
            "source": obs_msg.source,
            "signature": obs_msg.signature,
        }

        strategy.add_observation(obs_msg.sender, obs)
```
I can see it being useful having a helper method for extracting observation from message.